client/js/analysisui/layoutvariablesview.js

```diff
@@ -2,70 +2,136 @@
+    this.ignoreTransform = true;
+    var cell = this.addLayout("supplier", 0, 0, false, this.supplierGrid);
+    this.ignoreTransform = false;
+    cell.horizontalStretchFactor = 0.5;
+    cell.dockContentWidth = true;
+    cell.dockContentHeight = true;
+    cell.spanAllRows = true;
+
+
+    this.addHeader = function(title) {
+        var cell = this.addCell(0, 0, false);
+    };
 
-    setInfo: function(resources, format, level) {
+
+    this.setInfo = function(resources, style, level) {
 
         this.resources = resources;
-        this.format = format;
+        this.style = style;
         this.level = level;
-
-        var columns = this.resources.columns;
-        for (var i = 0; i < columns.length; i++) {
-            var column = columns[i];
-
-            var c1 = this.supplierGrid.addCell(0, i, false, $('<div class="silky-variable silky-variable-type-' + column.measureType + '">' + column.name + '</div>'));
-            c1.horizontalStretchFactor = 1;
-            c1.dockContentWidth = true;
-            c1.clickable(true);
+        this._items = { _list:[] };
+
+        this.populateItemList();
+        this.renderItemList();
+    };
+
+    this.getItem = function(index) {
+        return this._items._list[index];
+    };
+
+    this.getSelectedItem = function(index) {
+        if (this.supplierGrid.selectedCellCount() > index)
+            return this.getItem(this.supplierGrid.getSelectedCell(index).data.row);
+
+        return null;
+    };
+
+    this.pullSelectedItem = function(index) {
+        var item = this.getSelectedItem(index);
+        if (item !== null)
+            item.used += 1;
+        return item;
+    };
+
+    this.pushItem = function(formatted) {
+        for (var i = 0; i < this._items._list.length; i++) {
+            var item = this._items._list[i];
+            if (item.value.equalTo(formatted)) {
+                item.used -= 1;
+                break;
+            }
         }
-    },
+    };
 
-    addTarget: function(target) {
+    this.addTarget = function(target) {
         this._targets.push(target);
         var self = this;
         target.targetGrid.on('layoutgrid.gotFocus', function() {
             self.supplier.clearSelection();
         });
-    }
+    };
+
+
+    this.populateItemList = function() {
+        var columns = this.resources.columns;
+        for (var i = 0; i < columns.length; i++) {
+            var column = columns[i];
+            var item = { value: new Formatdef.constructor(column.name, Formatdef.variable), used: 0, properties: { type: column.measureType } };
+            this._items._list.push(item);
+            this._items[column.name] = item;
+        }
+    };
+
+    this.renderItemList = function() {
+        for (var i = 0; i < this._items._list.length; i++) {
+            var item = this._items._list[i];
+            this['render_' + item.value.format.name](item, i);
+        }
+    };
+
+    this.render_variable = function(item, row) {
```

Comment:
This function is only called by name, as in this["render_" + type]. So i have used the underscore to emphasize this use. I can change it to this[type + "Render"].